go/runtime/host/tdx/cid_pool.go

```diff
@@ -0,0 +1,55 @@
+package tdx
+
+import (
+	"fmt"
+	"sync"
+)
+
+// CidPool is a pool of CIDs that can be assigned to VMs.
+type CidPool struct {
+	l sync.Mutex
+
+	free map[uint32]struct{}
+	used map[uint32]struct{}
+}
+
+// NewCidPool creates a new CID pool containing the given range of CIDs.
+func NewCidPool(start, count uint32) (*CidPool, error) {
+	if start < 10 {
+		return nil, fmt.Errorf("CID identifiers between 0 and 10 are reserved")
+	}
+
+	c := CidPool{
+		free: make(map[uint32]struct{}),
+		used: make(map[uint32]struct{}),
+	}
+	for cid := start; cid < start+count; cid++ {
+		c.free[cid] = struct{}{}
```

Comment:
If `count` is big, you are wasting memory here. I believe that there will be little used CIDs, so I would rather have only one map `used` and change `Allocate` method to iterate until it finds one that is not used.
If you decide to stay with this approach, you can also remove `used` and solve this with `free`, `start` and `count`.

Reply:
I know, initially I had the more complicated variant implemented, but then just simplified it like this. I don't think this will contain a huge amount of entries in practice as running even 1024 virtual machines on a single host would probably not work very well.

And for 1024 CIDs this takes around 4KiB + overhead.